sycl/include/sycl/ext/oneapi/experimental/device_architecture.hpp

```diff
@@ -24,92 +28,116 @@ enum class architecture {
   //   - the unique ID of the new architecture in SYCL RT source code to support
   //     querying the device architecture
   //
-  x86_64,
-  intel_cpu_spr,
-  intel_cpu_gnr,
-  intel_gpu_bdw,
-  intel_gpu_skl,
-  intel_gpu_kbl,
-  intel_gpu_cfl,
-  intel_gpu_apl,
+  x86_64 = 0x00000000,
+  //
+  // Intel CPU architectures
+  //
+  // The requirement for the unique ID for intel_cpu_* architectures below is:
+  // - the ID must start with 0x0 (to avoid the integer overflow)
```

Comment:
I don't understand the the comment about "to avoid the integer overflow".  Can't we just change the type of the enumeration to be:

```
enum class architecture : uint32_t
```

Reply:
Oh, right, we have `enum class architecture : /* unspecified */` in the spec.